src/ui/server/helpers/required-fields/declarations/index.ts

```diff
@@ -31,16 +41,44 @@ export const getAntiBriberyCodeOfConductTasks = (hasAntiBriberyCodeOfConduct?: b
   return [HAS_ANTI_BRIBERY_CODE_OF_CONDUCT];
 };
 
+/**
+ * getModernSlaveryTasks
+ * Get "modern slavery" tasks depending on the provided answers
+ * @param {ApplicationDeclarationModernSlavery} modernSlavery: Application "Modern slavery" answers
+ * @returns {Array} Modern slavery tasks
+ */
+export const getModernSlaveryTasks = (modernSlavery: ApplicationDeclarationModernSlavery) => {
+  const tasks = [WILL_ADHERE_TO_ALL_REQUIREMENTS, HAS_NO_OFFENSES_OR_INVESTIGATIONS, IS_NOT_AWARE_OF_EXISTING_SLAVERY];
+
+  if (modernSlavery.willAdhereToAllRequirements === false) {
```

Comment:
Switch-case please.

Reply:
I do you envison a switch/case here? The only way to do it would be like this:

```ts
switch (modernSlavery.willAdhereToAllRequirements) {
  case false:
    tasks.push(CANNOT_ADHERE_TO_ALL_REQUIREMENTS);
}

switch (modernSlavery.hasNoOffensesOrInvestigations) {
  case false:
    tasks.push(CANNOT_ADHERE_TO_ALL_REQUIREMENTS);
}

switch (modernSlavery.isNotAwareOfExistingSlavery) {
  case false:
    tasks.push(CANNOT_ADHERE_TO_ALL_REQUIREMENTS);
}
```

Which seems pointless / code smell, as it's a single `switch` and a single `case` for 3x different fields with individual conditions.

The if statements make a lot more sense IMO. If this was only for one field then i'd agree a switch/case would be cleaner.

Reply:
Thanks for the explanation.